### PR TITLE
Regenerate with dust 0.9.10

### PR DESCRIPTION
**Pull request by richfitz**
Requires https://github.com/mrc-ide/dust/pull/246
Requires https://github.com/mrc-ide/mcstate/pull/137

Straight regeneration, plus two patched tests

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -73,8 +73,8 @@ RoxygenNote: 7.1.1
 Roxygen: list(markdown = TRUE)
 SystemRequirements: C++11
 Remotes:
-    mrc-ide/dust,
+    mrc-ide/dust@i245-shared-int,
```
Comment (johnlees) [inline DESCRIPTION:76]:
```suggestion
    mrc-ide/dust,
```

File: DESCRIPTION [hunk 2]
```diff
@@ -73,8 +73,8 @@ RoxygenNote: 7.1.1
...
     mrc-ide/eigen1,
-    mrc-ide/mcstate,
+    mrc-ide/mcstate@dust-0.9.10,
```
Comment (johnlees) [inline DESCRIPTION:78]:
```suggestion
    mrc-ide/mcstate,
```

File: docker/Dockerfile [hunk 3]
```diff
@@ -22,5 +22,6 @@ RUN install2.r --error \
         socialmixr
 
 RUN Rscript -e 'remotes::install_github(c("mrc-ide/dust",  "mrc-ide/mcstate", "mrc-ide/eigen1", "mrc-ide/odin", "mrc-ide/odin.dust"))'
+RUN Rscript -e 'remotes::install_github(c("mrc-ide/dust@i245-shared-int", "mrc-ide/mcstate@dust-0.9.10"))'
```
Comment (johnlees) [inline docker/Dockerfile:25]: Codefactor complains about this, but I think it's going away anyway
```suggestion
```

**Review by johnlees — APPROVED** [review]

File: docker/Dockerfile (hunk 3 above)
Comment (richfitz) [reply docker/Dockerfile:25]: yeah, that's an expected complaint, will remove once everything else is merged (having a mare with mcstate....)

**Pull request merged**